public/app/features/admin/state/reducers.ts

```diff
@@ -201,8 +203,52 @@ export const { usersFetched, usersFetchBegin, usersFetchEnd, queryChanged, pageC
   userListAdminSlice.actions;
 export const userListAdminReducer = userListAdminSlice.reducer;
 
+// UserListAnonymousPage
+
+const initialUserListAnonymousDevicesState: UserListAnonymousDevicesState = {
+  devices: [],
+  query: '',
+  page: 0,
+  perPage: 50,
+  totalPages: 1,
+  showPaging: false,
+  filters: [{ name: 'activeLast30Days', value: false }],
+  isLoading: true,
+};
+
+interface UsersAnonymousDevicesFetched {
+  devices: UserAnonymousDeviceDTO[];
+  currentPage: number;
+  perPage: number;
+  totalCount: number;
+}
+
+export const userListAnonymousDevicesSlice = createSlice({
+  name: 'userListAnonymousDevices',
+  initialState: initialUserListAnonymousDevicesState,
+  reducers: {
+    usersAnonymousDevicesFetched: (state, action: PayloadAction<UsersAnonymousDevicesFetched>) => {
+      const { devices } = action.payload;
+      const totalPages = 1;
+      const perPage = 10;
+      return {
+        ...state,
+        totalPages,
+        perPage,
+        devices,
+        showPaging: totalPages > 1,
+        isLoading: false,
+      };
+    },
+  },
```

Comment:
```suggestion
      return {
        ...state,
        devices,
        isLoading: false,
      };
    },
  },
```
You can remove all the extra fields from here as well (and from the state type).